### PR TITLE
Add r2_score based on requested changes.

### PR DESCRIPTION
**Pull request by alexbispo**
This PR is a continuation of the PR #72. 

I applied the requested changes.

### CODE REVIEW

File: lib/scholar/metrics.ex [hunk 1]
```diff
@@ -941,6 +942,31 @@ defmodule Scholar.Metrics do
     auc(fpr, tpr)
   end
 
+   @doc ~S"""
```
Comment (josevalim) [inline lib/scholar/metrics.ex:945]:
```suggestion
  @doc ~S"""
```

File: lib/scholar/metrics.ex [hunk 2]
```diff
@@ -941,6 +942,31 @@ defmodule Scholar.Metrics do
...
+      >
+  """
+  defn r2_score(y_true, y_pred) do
+    check_shape(y_true, y_pred)
+    ssr = squared_euclidean(y_true, y_pred)
+
+    y_mean = Nx.broadcast(Nx.mean(y_true), Nx.shape(y_true))
+    sst = squared_euclidean(y_true, y_mean)
+
+    1 - ssr/sst
```
Comment (josevalim) [inline lib/scholar/metrics.ex:967]:
```suggestion
    1 - ssr / sst
```

**Comment by msluszniak** [comment]
Thanks for a PR. I think that for now, we don't need to implement `sample_weight` option, because I will make a PR which adds them to all metrics when they will be implemented. However, I think that it is worth to add similar option to sci-kit r2: `force_finite`. What do you think about it?

**Review by msluszniak — APPROVED** [review]

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**